CommonTools/TriggerUtils/src/GenericTriggerEventFlag.cc

```diff
@@ -402,28 +426,32 @@ bool GenericTriggerEventFlag::acceptL1( const edm::Event & event, const edm::Eve
   // An empty L1 logical expressions list acts as switch.
   if ( ! onL1_ || l1LogicalExpressions_.empty() ) return ( ! andOr_ ); // logically neutral, depending on base logical connective
 
-  // Getting the L1 event setup
-  l1Gt_->getL1GtRunCache( event, setup, useL1EventSetup, useL1GtTriggerMenuLite ); // FIXME This can possibly go to initRun()
-
   // Determine decision of L1 logical expression combination and return
   if ( andOrL1_ ) { // OR combination
     for ( std::vector< std::string >::const_iterator l1LogicalExpression = l1LogicalExpressions_.begin(); l1LogicalExpression != l1LogicalExpressions_.end(); ++l1LogicalExpression ) {
-      if ( acceptL1LogicalExpression( event, *l1LogicalExpression ) ) return true;
+      if ( acceptL1LogicalExpression( event, setup, *l1LogicalExpression ) ) return true;
     }
     return false;
   }
   for ( std::vector< std::string >::const_iterator l1LogicalExpression = l1LogicalExpressions_.begin(); l1LogicalExpression != l1LogicalExpressions_.end(); ++l1LogicalExpression ) {
-    if ( ! acceptL1LogicalExpression( event, *l1LogicalExpression ) ) return false;
+    if ( ! acceptL1LogicalExpression( event, setup, *l1LogicalExpression ) ) return false;
   }
   return true;
 
 }
 
 
 /// Was this event accepted by this particular L1 algorithms' logical expression?
-bool GenericTriggerEventFlag::acceptL1LogicalExpression( const edm::Event & event, std::string l1LogicalExpression )
+bool GenericTriggerEventFlag::acceptL1LogicalExpression( const edm::Event & event, const edm::EventSetup & setup, std::string l1LogicalExpression )
 {
 
+  // Getting the L1 event setup
+  if ( stage2_ )
+    l1uGt_->retrieveL1(event,setup);
+  else
+    l1Gt_->getL1GtRunCache( event, setup, useL1EventSetup, useL1GtTriggerMenuLite ); // FIXME This can possibly go to initRun()
```

Comment:
nope :(
but I think I looked for an analogous function in the L1GlobalUtils, but I did not find it
has it been developed in the meantime ?
I would be happy in updating the code ....